src/python/WMCore/REST/Format.py

```diff
@@ -612,6 +613,6 @@ def stream_maybe_etag(size_limit, etag, reply):
 
     # OK, respond with the buffered reply as a plain string.
     res.headers['Content-Length'] = size
-    result = "".join(result)
+    result = b"".join(result)
```

Comment:
Dario, just a question out of curiosity. Would you remember or still have the error that we were getting without this conversion to byte string?

Reply:
This is a great question. I have to admit I do not remember why I added this change (and i did not take note of this). I think it's just quicker if i remove this change and we see what jenkins has to say

Reply:
Without this change, the unittest `test/python/WMCore_t/ReqMgr_t/Service_t/Auxiliary_t.py:AuxiliaryTest.testInfo` (and also other from `Auxiliary_t.py`) fails with

```plaintext
cherrypy.error: INFO: [15/Jun/2021:16:25:23]  SERVER OTHER ERROR __builtins__.TypeError 7afe8c458b6fe6294785b341006200c8 (sequence item 0: expected str instance, bytes found)
cherrypy.error: INFO: [15/Jun/2021:16:25:23]    Traceback (most recent call last):
cherrypy.error: INFO: [15/Jun/2021:16:25:23]      File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/REST/Server.py", line 726, in default
cherrypy.error: INFO: [15/Jun/2021:16:25:23]        return self._call(RESTArgs(list(args), kwargs))
cherrypy.error: INFO: [15/Jun/2021:16:25:23]      File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/REST/Server.py", line 842, in _call
cherrypy.error: INFO: [15/Jun/2021:16:25:23]        return stream_maybe_etag(apiobj.get('etag_limit', self.etag_limit), etagger, reply)
cherrypy.error: INFO: [15/Jun/2021:16:25:23]      File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/REST/Format.py", line 616, in stream_maybe_etag
cherrypy.error: INFO: [15/Jun/2021:16:25:23]        result = "".join(result)
cherrypy.error: INFO: [15/Jun/2021:16:25:23]    TypeError: sequence item 0: expected str instance, bytes found
cherrypy.access: INFO: [15/Jun/2021:16:25:23] 028e0d827607 127.0.0.1 "GET /reqmgr2/data/info HTTP/1.1" 500 Internal Server Error [data: 198 in 2661 out 90484 us ] [auth: - "" "" ] [ref: "" "WMCore.Services.Requests/v002" ]
```

Reply:
So, we can investigate why we have such bytes moving around, but i am not sure that we can simplify anything here and i would just keep this change for the time being. What do you think?

Reply:
Yes, I agree. I would suggest though to put a FIXME or TODO comment saying that for some reason `result` is a byte string and it fails in python3, such that we can investigate it in the future and try to move to sequence/normal strings.
If we want to be extremely conservative, we could even do something like:
if PY3:
  then do the byte thing, otherwise don't do anything 
else:
  previous code.

Anything is good with me.

Reply:
I added a TODO and an `if PY3` to that line.

(With the risk of repeating myself and stating obvious facts, in py2 `b"hello"` is the same as `"hello"`, so we added the `if PY3` for the sole purpose of easily getting back to this change)